lib/src/pgn.dart

```diff
@@ -0,0 +1,831 @@
+          initHeaders)
+      .parse(pgn);
+  return games;
+}
+
+/// Represents the color of a comment
+///
+/// Can be green, red, yellow, and blue
+enum CommentShapeColor {
+  green,
+  red,
+  yellow,
+  blue;
+
+  String get string {
+    switch (this) {
+      case CommentShapeColor.green:
+        return 'Green';
+      case CommentShapeColor.red:
+        return 'Red';
+      case CommentShapeColor.yellow:
+        return 'Yellow';
+      case CommentShapeColor.blue:
+        return 'Blue';
+    }
+  }
+
+  static CommentShapeColor? parseShapeColor(String str) {
+    switch (str) {
+      case 'G':
+        return CommentShapeColor.green;
+      case 'R':
+        return CommentShapeColor.red;
+      case 'Y':
+        return CommentShapeColor.yellow;
+      case 'B':
+        return CommentShapeColor.blue;
+      default:
+        return null;
+    }
+  }
+}
+
+/// A comment shape
+///
+/// Example of a comment shape "%cal Ra1b2" with color: Red from:a1 to:b2
+@immutable
+class PgnCommentShape {
+  final CommentShapeColor color;
+  final Square from;
+  final Square to;
+
+  const PgnCommentShape(
+      {required this.color, required this.from, required this.to});
+
+  @override
+  String toString() {
+    return to == from
+        ? '${color.string[0]}${toAlgebraic(to)}'
+        : '${color.string[0]}${toAlgebraic(from)}${toAlgebraic(to)}';
+  }
+
+  /// Parse the str for any comment or return null
+  static PgnCommentShape? fromPgn(String str) {
+    final color = CommentShapeColor.parseShapeColor(str.substring(0, 1));
+    final from = parseSquare(str.substring(1, 3));
+    if (color == null || from == null) return null;
+    if (str.length == 3) {
+      return PgnCommentShape(color: color, from: from, to: from);
+    }
+    final to = parseSquare(str.substring(3, 5));
+    if (str.length == 5 && to != null) {
+      return PgnCommentShape(color: color, from: from, to: to);
+    }
+    return null;
+  }
+}
+
+/// Represents the type of [PgnEvaluation]
+///
+/// Can of of type pawns or mate
+enum EvalType { pawns, mate }
+
+/// A class containing an evaluation
+/// A Evaluation object can be created used .pawns or .mate contructor
+@immutable
+class PgnEvaluation {
+  final double? pawns;
+  final int? mate;
+  final int? depth;
+  final EvalType evalType;
+
+  /// Constructor to create Evaluation of type pawns
+  const PgnEvaluation.pawns(
+      {required this.pawns,
+      this.depth,
+      this.mate,
+      this.evalType = EvalType.pawns});
+
+  /// Constructor to create Evaluation of type mate
+  const PgnEvaluation.mate(
+      {required this.mate,
+      this.depth,
+      this.pawns,
+      this.evalType = EvalType.mate});
+
+  @override
+  bool operator ==(Object other) =>
+      other is PgnEvaluation &&
+      pawns == other.pawns &&
+      depth == other.depth &&
+      mate == other.mate;
+
+  @override
+  int get hashCode => evalType == EvalType.pawns
```

Comment:
curious implementation here, I prefer the straightforward:
```
Object.hash(pawns, mate, depth, evalType)
```
I know we should never write such code, I should probably have used `freezed` in that project too.